src/connection.cpp

```diff
@@ -269,14 +269,16 @@ void Connection::connect() {
 }
 
 bool Connection::write(const RequestCallback::Ptr& callback, bool flush_immediately) {
-  int32_t result = internal_write(callback, flush_immediately);
+  int32_t result = internal_write(callback, flush_immediately, compressor_.get());
```

Comment:
A compression instance already exists in this class and should be called directly; extra `compression` parameter added to methods in this class and not needed and can be reverted.